### PR TITLE
feat(mock): add arrayMin and arrayMax mock override options

### PR DESCRIPTION
**Pull request by cowofevil**
## Status

**WIP**

## Description

Add the ability to override the minimum and maximum length of arrays generated by faker for properties that specify multiple items. This can be handy for a situation where you want to use the provided data generators for unit testing and need to guaranteed a minimum number of items in an array larger than 1.

## Todos

- [x] Tests
- [x] Documentation
- [ ] Changelog Entry (unreleased)

### CODE REVIEW

**Comment by cowofevil** [comment]
@anymaniax I'm not sure how best to test this change and would like some guidance. I checked out the current unit test suite and it seemed rather incomplete compared to the examples. I updated the basic react query sample just to demonstrate the effect of this new feature. Let me know how I should test this and I'll update the PR.

File: src/core/generators/mocks.ts [hunk 1]
```diff
@@ -84,18 +86,27 @@ const getMockWithoutFunc = (
...
+        ? `Array.from({length: faker.datatype.number({` +
+            `min: ${mockOptionsWithoutFunc?.arrayMin ?? 1}, ` +
+            `max: ${mockOptionsWithoutFunc?.arrayMax ?? 10}}` +
+            `)}, () => faker.datatype.number())`
         : 'faker.datatype.number().toString()';
     case 'string':
       return isArray
-        ? `Array.from({length: faker.datatype.number({min: 1, max: 10})}, () => faker.random.word())`
+        ? `Array.from({length: faker.datatype.number({` +
+            `min: ${mockOptionsWithoutFunc?.arrayMin ?? 1},` +
```
Comment (anymaniax) [inline src/core/generators/mocks.ts]: can you put the default values in the normalizer of the options?

Comment (cowofevil) [reply src/core/generators/mocks.ts]: Fixed.

**Comment by anymaniax** [comment]
You can also update a specification in the tests folder to add a test at generation time

**Comment by cowofevil** [comment]
@anymaniax Moved the feature demonstration from the 'samples' folder to the 'tests' folder.

**Comment by cowofevil** [comment]
@anymaniax If you don't have anymore feedback about the PR, would you like me to squash the commits and update the branch for merge?

**Comment by cowofevil** [comment]
@anymaniax Ping. Are you interested in this PR or would you rather me close it?

**Comment by anymaniax** [comment]
Hello @cowofevil, sorry I am still interested but wanted to fix other things before merging this. I will take a look to it this weekend

**Review by anymaniax — APPROVED** [review]

**Pull request merged**